cachito/web/content_manifest.py

```diff
@@ -377,16 +403,23 @@ def to_purl(self):
                 return f"pkg:pypi/{name}@{self.version}"
             elif parsed_url.scheme.startswith("git+"):
                 # Version is git+<git_url>
-                scheme = parsed_url.scheme[len("git+") :]
-                vcs_url = f"{scheme}://{parsed_url.netloc}{parsed_url.path}"
-                repo_url, ref = vcs_url.rsplit("@", 1)
-                return self.to_vcs_purl(repo_url, ref)
+                return self.to_git_purl(parsed_url)
             else:
                 # Version is a plain URL
                 fragments = urllib.parse.parse_qs(parsed_url.fragment)
                 checksum = fragments["cachito_hash"][0]
                 quoted_url = urllib.parse.quote(self.version, safe="")
                 return f"pkg:generic/{name}?download_url={quoted_url}&checksum={checksum}"
+        elif self.type == "rubygems":
+            if self.kind == "PATH":
+                # Package is relative to the parent module
+                normpath = os.path.normpath(self.path)
+                return f"{PARENT_PURL_PLACEHOLDER}#{normpath}"
```

Comment:
Seeing this, I realized we may need more complex logic here :disappointed: 

If the parent package is not in the repo root, its purl will look something like this:
```
pkg:github/organization/repo@commit-hash#subpath
```
Appending `#{normpath}` would make the purl invalid
```
pkg:github/organization/repo@commit-hash#subpath#other-path
```
We would want
```
pkg:github/organization/repo@commit-hash#subpath/other-path
```
The only reason this works for gomod is that the parent purls never contain subpaths (top-level purls can use `pkg:golang` unlike other package managers)

Reply:
Or maybe the more complex logic will need to go to the code which replaces the parent placeholder

Reply:
You're right, I added a test case for this. The path that is in the dependency metadata is the path relative to the repo root, so the parent purl is now generated from the `request.repo`.

Reply:
> The path that is in the dependency metadata is the path relative to the repo root

Hmm, I think we want it to be relative to the package

Reply:
Or maybe it's fine, just needs to be documented

Reply:
In the interest of keeping Cachito consistent, I would prefer for the `version` of PATH deps to be relative to the package dir, not to the repository root dir. In the content manifest code, you would then have to get the full path from the parent package path + the dependency path

Reply:
If we want `version` to contain relative path to parent package directory, then the function generating PURL has to get information about relative path from the parent package to the repo root. I have no better idea then to get this from package name (we could have `resolve_rubygems` function to return package path as part of the metadata it returns, if we wanted), so this PR depends on merging #692. What do you think about [New way of handling rubygems PURL](https://github.com/containerbuildsystem/cachito/pull/684/commits/8374d81b97d749d3d5c7fb3ad9e9979eeac360ef)?

Reply:
It's not completely finished, I'll need to change the function generating PATH metadata, I'm mostly interested if I'm going in the right direction with PURL generation.

Reply:
> the function generating PURL has to get information about relative path from the parent package to the repo root

https://github.com/containerbuildsystem/cachito/blob/8374d81b97d749d3d5c7fb3ad9e9979eeac360ef/cachito/web/content_manifest.py#L171

The `package` object should have a `path` - which is exactly the relative path from the repo root to the package (unless the package is in the repo root, in which case the path should be None)

Reply:
The updates to the purl module make sense to me :+1: